### PR TITLE
[earcut] Added types.

### PR DESCRIPTION
**Pull request by NaridaL**
Duplicated export mirrors actual source: https://github.com/mapbox/earcut/blob/master/src/earcut.js#L3-L4

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
PR not accepted https://github.com/mapbox/earcut/pull/89
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/earcut/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * @param vertices A flat array of vertice coordinates like [x0,y0, x1,y1, x2,y2, ...].
+ * @param holes An array of hole indices if any (e.g. [5, 8] for a 12-vertice input would mean one hole with vertices 5–7 and another with 8–11).
+ * @param dimensions The number of coordinates per vertice in the input array (2 by default).
+ * @return A flat array with each group of three numbers indexing a triangle in the `vertices` array.
+ * @example earcut([10,0, 0,50, 60,60, 70,10]); // returns [1,0,3, 3,2,1]
+ * @example with a hole: earcut([0,0, 100,0, 100,100, 0,100,  20,20, 80,20, 80,80, 20,80], [4]); // [3,0,4, 5,4,0, 3,4,7, 5,0,1, 2,3,7, 6,5,1, 2,7,6, 6,1,2]
+ * @example with 3d coords: earcut([10,0,1, 0,50,2, 60,60,3, 70,10,4], null, 3); // [1,0,3, 3,2,1]
+ */
+declare function earcut(vertices: ArrayLike<number>, holes?: ArrayLike<number>, dimensions?: number): number[];
```
Comment (weswigham) [inline types/earcut/index.d.ts]: I'd much prefer to see a definition styled like
```ts
interface EarcutStatic {
   (vertices: ArrayLike<number>, holes?: ArrayLike<number>, dimensions?: number): number[];
   default: EarcutStatic;
}
declare const exports: EarcutStatic;
export = exports;
export as namespace earcut;
```
There's also [`flatten`](https://github.com/mapbox/earcut/blob/master/src/earcut.js#L635) and [`deviation`](https://github.com/mapbox/earcut/blob/master/src/earcut.js#L598) methods (which would then be on `EarcutStatic`) which are likely worth including, but I'll still merge the PR if for some reason you don't want to add them.

**Review by weswigham — CHANGES_REQUESTED** [review]
Small style request, if you could.

**Comment by NaridaL** [comment]
Not convinced that this is better than `function earcut(); namespace earcut {}`, but sure :-P

**Comment by weswigham** [comment]
I think it reads better like this when there's an explicit `default` member and a callable top-level member. If the top-level was constructable I'd go for the namespace/class merge, too (otherwise the class instance type can be a pain to make).

**Review by weswigham — APPROVED** [review]

**Pull request merged**